dxr/plugins/core.py

```diff
@@ -308,6 +301,26 @@ def filter(self):
             raise BadTerm('Path globs need at least 3 literal characters in a row '
                           'for speed.')
 
+class FilenameFilter(Filter):
+    """Substring filter for file names"""
+    name = 'file'
+    domain = FILE
+    description = Markup('File to search within. <code>*</code>, '
+                         '<code>?</code>, and <code>[...]</code> act '
+                         'as shell wildcards.')
+
+    @negatable
+    def filter(self):
+        glob = self._term['arg']
+        try:
+            return es_regex_filter(
+                regex_grammar.parse(glob_to_regex(glob)),
+                'file_name',
```

Comment:
I'm still on the fence... the `name` and `description` vars also differ, as does the `BadTerm` text.  Seems like you'd be factoring out basically just the call to `es_regex_filter` (with some parameter for the mapping name)...  I'm leaning toward saying it's not worth it - what do you think?